core/templates/expressions/expression-syntax-tree.service.spec.ts

```diff
@@ -15,42 +15,77 @@
 /**
  * @fileoverview Unit tests for expression-syntax-tree.service.ts
  */
+import { TestBed } from '@angular/core/testing';
 
-// TODO(#7222): Remove the following block of unnnecessary imports once
-// the code corresponding to the spec is upgraded to Angular 8.
-import { UpgradedServices } from 'services/UpgradedServices';
-// ^^^ This block is to be removed.
-
-require('App.ts');
-require('expressions/expression-syntax-tree.service.ts');
+import { ExpressionSyntaxTreeService, ExpressionError,
+  ExprUndefinedVarError, ExprWrongArgTypeError,
+  ExprWrongNumArgsError } from
+  'expressions/expression-syntax-tree.service.ts';
 
 describe('Expression syntax tree service', () => {
-  beforeEach(angular.mock.module('oppia'));
-  beforeEach(angular.mock.module('oppia', function($provide) {
-    var ugs = new UpgradedServices();
-    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
-      $provide.value(key, value);
-    }
-  }));
-
   describe('expression syntax tree service', () => {
-    let ExpressionSyntaxTreeService = null;
+    let expressionSyntaxTreeService: ExpressionSyntaxTreeService;
 
-    beforeEach(angular.mock.inject(($injector) => {
-      ExpressionSyntaxTreeService =
-          $injector.get('ExpressionSyntaxTreeService');
-    }));
+    beforeEach(() => {
+      expressionSyntaxTreeService = TestBed.get(ExpressionSyntaxTreeService);
+    });
 
     it('should throw if environment is not found', () => {
-      expect(() => ExpressionSyntaxTreeService.lookupEnvs('', [])).toThrow();
+      expect(() => expressionSyntaxTreeService.lookupEnvs('', [])).toThrow();
     });
 
     it('should return the correct environment if exists', () => {
       const expected = 'bar';
-      const actual =
-          ExpressionSyntaxTreeService.lookupEnvs('foo', [{foo: 'bar'}]);
+      const actual = expressionSyntaxTreeService.lookupEnvs('foo', [
+        {foo: 'bar'}
+      ]);
 
       expect(expected).toBe(actual);
     });
   });
+
+  describe('ExpressionError', () => {
+    let expressionError: ExpressionError;
+
+    it('should extend Error object', () => {
+      expressionError = new ExpressionError();
+
+      expect(expressionError.name).toBe('ExpressionError');
+      expect(expressionError instanceof Error).toBe(true);
+    });
+  });
+
+  describe('ExprUndefinedVarError', () => {
+    let exprUndifinedVarError: ExprUndefinedVarError;
+
+    it('should extend ExpressionError class', () => {
+      const exampleVar = undefined;
+      exprUndifinedVarError = new ExprUndefinedVarError(exampleVar, []);
```

Comment:
Whoops almost missed this, there's a typo here, it should be `exprUndefinedVarError`